### ISSUE TITLE
GraalVM macOS build is not recognised by macOS Java tools

### ISSUE THREAD

**Issue by breun** [oracle/graal#801]
I have GraalVM 1.0.0-rc9 installed on macOS 10.14.1:

```
/Library/Java/JavaVirtualMachines $ ls -1
graalvm-ce-1.0.0-rc9
openjdk10
openjdk11
openjdk8
```

However, GraalVM is not recognised as a JDK by `/usr/libexec/java_home`, which for instance can list JDK's and run commands on a specific JDK:

```
/Library/Java/JavaVirtualMachines $ /usr/libexec/java_home --verbose
Matching Java Virtual Machines (3):
    11.0.1, x86_64:	"OpenJDK 11.0.1"	/Library/Java/JavaVirtualMachines/openjdk11/Contents/Home
    10.0.2, x86_64:	"OpenJDK 10.0.2"	/Library/Java/JavaVirtualMachines/openjdk10/Contents/Home
    1.8.0_192, x86_64:	"OpenJDK 8"	/Library/Java/JavaVirtualMachines/openjdk8/Contents/Home

/Library/Java/JavaVirtualMachines/openjdk11/Contents/Home
```

The cause seems to be that there is no `Contents/Info.plist` file included in the GraalVM build for macOS:

```
/Library/Java/JavaVirtualMachines $ find . -name Info.plist
./openjdk8/Contents/Info.plist
./openjdk11/Contents/Info.plist
./openjdk10/Contents/Info.plist
```

I think it would be a good idea to also add a `Contents/Info.plist` file to the GraalVM macOS distribution.

**Comment by dougxc** [comment]
GraalVM EE has the expected "extra" files:
```
ls -l /Library/Java/JavaVirtualMachines/graalvm-ee-1.0.0-rc9/Contents /Library/Java/JavaVirtualMachines/graalvm-ee-1.0.0-rc9/Contents/MacOS/
/Library/Java/JavaVirtualMachines/graalvm-ee-1.0.0-rc9/Contents:
total 8
drwxr-xr-x  17 root  staff   544 Nov  5 19:02 Home
-rw-r--r--@  1 root  wheel  1581 Oct  6 18:57 Info.plist
drwxr-xr-x   3 root  staff    96 Nov  5 19:02 MacOS

/Library/Java/JavaVirtualMachines/graalvm-ee-1.0.0-rc9/Contents/MacOS/:
total 0
lrw-r--r--@ 1 root  wheel  32 Nov  2 07:06 libjli.dylib -> ../Home/jre/lib/jli/libjli.dylib
```
These are copied from an existing Oracle JDK8.

We build our own OpenJDK8 binary on macOS and so I'm not sure where we should source the contents of `Info.plist` from.

**Comment by breun** [comment]
I don't know what the contents of `Info.plist` for GraalVM EE are, but I believe an `Info.plist` file could just be added setting whatever version, vendor, etc. values are relevant for GraalVM. These values will also be used in the output of `/usr/libexec/java_home -V` to identify the JDK.

**Comment by dougxc** [comment]
I'll copy the entries from GraalVM EE and they look applicable.

**Comment by breun** [comment]
I guess it would be good if it's possible to distinguish between CE and EE based on the output of `/usr/libexec/java_home -V`.

**Comment by dougxc** [comment]
Sorry, this was a regression in the build scripts for the JDK on which GraalVM is built. It will be fixed in RC13. Hopefully, the GraalVM specific identification will also be in Info.plist by then (/cc @ansalond ).

**Comment by ansalond** [comment]
Sorry for the long delay, I finally addressed this issue.

In RC13, for GraalVM CE we will have:
```
<key>CFBundleName</key>
<string>GraalVM CE 1.0.0-rc13</string>
```

Example:
```
$ /usr/libexec/java_home -V
Matching Java Virtual Machines (...):
1.8.0_192, x86_64: "GraalVM CE 1.0.0-rc13" /Library/Java/JavaVirtualMachines/graalvm-ce-1.0.0-rc13/Contents/Home
[...]
```

**Comment by ryankennedy** [comment]
Related, the version number conflicts with the “vanilla” JDK, which makes it impossible (AFAICT, maybe there’s another parameter I could be using) to use `/usr/libexec/java_home` to select GraalVM from among the installed JVMs…

```$ /usr/libexec/java_home -V
Matching Java Virtual Machines (3):
    11.0.2, x86_64:	"Java SE 11.0.2"	/Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk/Contents/Home
    1.8.0_202, x86_64:	"Java SE 8"	/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home
    1.8.0_202, x86_64:	"GraalVM CE 1.0.0-rc14"	/Library/Java/JavaVirtualMachines/graalvm-ce-1.0.0-rc14/Contents/Home

/Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk/Contents/Home
$ /usr/libexec/java_home -v 1.8.0_202
/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home```

I don’t know the plist details well enough to know if it’s possible to differentiate GraalVM via the version number exposed, but figured I’d mention it here.

**Comment by brentcrammond** [comment]
It looks like the JDK 11 in 1.9.3 is again not recognised by MacOS
/usr/libexec/java_home -V

It seems to be missing the MacOS directory, but there may be some other items that stop it being recognised.  Note the 1.9.3 JDK 8 is visible.

```
/usr/libexec/java_home -V
Matching Java Virtual Machines (4):
    13.0.1, x86_64:	"AdoptOpenJDK 13"	/Library/Java/JavaVirtualMachines/adoptopenjdk-13.jdk/Contents/Home
    11.0.5, x86_64:	"AdoptOpenJDK 11"	/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
    1.8.0_232, x86_64:	"AdoptOpenJDK 8"	/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home
    1.8.0_231, x86_64:	"GraalVM CE 19.3.0"	/Library/Java/JavaVirtualMachines/graalvm-ce-java8-19.3.0/Contents/Home

/Library/Java/JavaVirtualMachines/adoptopenjdk-13.jdk/Contents/Home
```

```
ls -l
total 0
drwxr-xr-x  3 root  wheel  96  4 Jul 11:01 adoptopenjdk-11.jdk
drwxr-xr-x  3 root  wheel  96  8 Nov 20:10 adoptopenjdk-13.jdk
drwxr-xr-x  3 root  wheel  96  7 Jun 14:22 adoptopenjdk-8.jdk
**drwxr-xr-x  3 root  wheel  96 25 Nov 10:42 graalvm-ce-java11-19.3.0**
drwxr-xr-x  3 root  wheel  96 25 Nov 10:03 graalvm-ce-java8-19.3.0

```

**Comment by ansalond** [comment]
Hi @brentcrammond, thanks for the report and sorry for the trouble. I just merged a fix to master, which will be included in the next release.

For the moment, a possible workaround is to:
1. extract the GraalVM archive to `/Library/Java/JavaVirtualMachines/`
2. `mkdir /Library/Java/JavaVirtualMachines/graalvm-ce-java11-19.3.0/Contents/MacOS`
3. `cd /Library/Java/JavaVirtualMachines/graalvm-ce-java11-19.3.0/Contents/MacOS`
4. `ln -s ../Home/lib/jli/libjli.dylib`

as originally suggested here https://twitter.com/ryenus/status/1197354320804634624

Other workarounds can be found on our public Slack:
https://graalvm.slack.com/archives/CNBFR78F9/p1574361681204900

**Comment by brentcrammond** [comment]
Hi Danilo

It looks like the fix to the master branch might not have worked. I understand the work around and will apply it for my testing, but it would be good to get this fixed, as I think there are a lot of users waiting to use it with Quarkus 1.2 when it is released.

I have downloaded the graalvm-ce-java11-darwin-amd64-19.3.0.2.tar.gz amd extracted to the JavaLibrary directory as normal, but I see that it still does not contain the MacOS directory.

![Screen Shot 2020-01-07 at 2 52 48 PM](https://user-images.githubusercontent.com/6261538/71862102-9e8d4780-315d-11ea-979e-e3103851f0b4.png)
![Screen Shot 2020-01-07 at 2 52 36 PM](https://user-images.githubusercontent.com/6261538/71862101-9df4b100-315d-11ea-8b7a-d9de044d6782.png)

**Comment by ansalond** [comment]
Hi Brent,
19.3.0.2 is a bugfix release for 19.3.0 that includes only a [small set of fixes](https://www.graalvm.org/docs/release-notes/19_3#19302) and that was not part of [our roadmap](https://www.graalvm.org/docs/release-notes/version-roadmap/).
The [fix for this issue](https://github.com/oracle/graal/commit/6d08523d2a6505d0e5f8ad8a77a3cf6d241f3a97) will be included in GraalVM 19.3.1, scheduled for Jan. 14.
Sorry for the misunderstanding, I should have been more clear.
